src/init-action-post-helper.ts

```diff
@@ -186,7 +188,7 @@ export async function run(
   // appropriate permissions.
   if (
     process.env["CODEQL_ACTION_EXPECT_UPLOAD_FAILED_SARIF"] === "true" &&
-    repositoryNwo.owner !== "github"
+    github.context.payload.pull_request?.head.repo.fork === false
```

Comment:
Nice. That's easy. Though, this will also fail to run for `push` and `workflow_dispatch` triggers. I think that neither of these would ever run in a fork.

So, this should be more generic since it also captures non-pr triggered runs:

```suggestion
    !github.context.payload.pull_request?.head.repo.fork
```

Reply:
And now that I think about it a little more, can you restructure this so that there is a log message emitted explaining that we don't delete the SARIF on forks when `CODEQL_ACTION_EXPECT_UPLOAD_FAILED_SARIF` is true?

```javascript
if (process.env["CODEQL_ACTION_EXPECT_UPLOAD_FAILED_SARIF"] === "true") {    
  if (github.context.payload.pull_request?.head.repo.fork === false) {
    // do delete
  } else {
    // log
}
```

Reply:
That makes sense 👍 the log message emitted also means I can remove the code comment ✨ 

I've pushed the changes here and to the non-fork PR. I'll re-request review once the checks pass as expected on both.